src/api/app/models/kiwi/repository.rb

```diff
@@ -44,7 +45,8 @@ def source_path_format
     return if source_path =~ /^(dir|iso|smb|this):\/\/.+/
     return if source_path =~ /\A#{URI.regexp(['ftp', 'http', 'https', 'plain'])}\z/
     if source_path =~ /^obs:\/\/([^\/]+)\/([^\/]+)$/
-      return if Project.valid_name?(Regexp.last_match(1)) && Project.valid_name?(Regexp.last_match(2))
+      return if Project.valid_name?(Regexp.last_match(1)) && Project.valid_name?(Regexp.last_match(2)) && repo_type == 'rpm-md'
```

Comment:
[nitpick] Codeclimate reports complex method, maybe we can split it?

Reply:
Maybe we can move it into a new method :thinking: